metadata-io/src/main/java/com/linkedin/metadata/entity/EntityServiceImpl.java

```diff
@@ -1293,60 +1061,59 @@ public Map<Urn, Entity> getEntities(@Nonnull final Set<Urn> urns, @Nonnull Set<S
   }
 
   @Override
-  public void produceMetadataAuditEvent(@Nonnull final Urn urn, @Nonnull final String aspectName,
-      @Nullable final RecordTemplate oldAspectValue, @Nullable final RecordTemplate newAspectValue,
-      @Nullable final SystemMetadata oldSystemMetadata, @Nullable final SystemMetadata newSystemMetadata,
-      @Nullable final MetadataAuditOperation operation) {
-    log.debug(String.format("Producing MetadataAuditEvent for ingested aspect %s, urn %s", aspectName, urn));
-    if (aspectName.equals(getKeyAspectName(urn))) {
-      produceMetadataAuditEventForKey(urn, newSystemMetadata);
-    } else {
-      final Snapshot newSnapshot = buildSnapshot(urn, newAspectValue);
-      Snapshot oldSnapshot = null;
-      if (oldAspectValue != null) {
-        oldSnapshot = buildSnapshot(urn, oldAspectValue);
-      }
-      _producer.produceMetadataAuditEvent(urn, oldSnapshot, newSnapshot, oldSystemMetadata, newSystemMetadata,
-          operation);
-    }
+  public Pair<Future<?>, Boolean> alwaysProduceMCLAsync(@Nonnull final Urn urn, @Nonnull final AspectSpec aspectSpec,
```

Comment:
Introducing some naming here, 2 key points. We have another path which will conditionally produce MCLs based on the aspect, flags, etc. Lets be clear in that using this doesn't follow that path. Second, and probably more important is the communication to the caller that these are *Async*. There is a future in the from the kafka producer that should be used and not dropped.